test/e2e/services.go

```diff
@@ -96,16 +96,60 @@ func startServicesForRules(t *testing.T, e e2e.Environment) (metricsRulesEndpoin
 	testutil.Ok(t, e2e.StartAndWaitReady(runnable))
 
 	createRulesYAML(t, e, bucket, runnable.InternalEndpoint(e2edb.AccessPortName), e2edb.MinioAccessKey, e2edb.MinioSecretKey)
+
 	rulesBackend := newRulesBackendService(e)
 	testutil.Ok(t, e2e.StartAndWaitReady(rulesBackend))
 
-	return rulesBackend.InternalEndpoint("http")
+	createLokiYAML(t, e, e2edb.MinioAccessKey, e2edb.MinioSecretKey, runnable.InternalEndpoint(e2edb.AccessPortName), bucket)
+
+	loki := newLokiService(e)
+	testutil.Ok(t, e2e.StartAndWaitReady(loki))
+
+	return rulesBackend.InternalEndpoint("http"), loki.InternalEndpoint("http")
 }
 
 func startServicesForLogs(t *testing.T, e e2e.Environment) (
 	logsEndpoint string,
 	logsExtEndpoint string,
 ) {
+
+	// Create S3 replacement for rules backend
+	bucket := "loki_test"
+	userID := strconv.Itoa(os.Getuid())
+	ports := map[string]int{e2edb.AccessPortName: 8090}
+	envVars := []string{
+		"MINIO_ROOT_USER=" + e2edb.MinioAccessKey,
+		"MINIO_ROOT_PASSWORD=" + e2edb.MinioSecretKey,
+		"MINIO_BROWSER=" + "off",
+		"ENABLE_HTTPS=" + "0",
+		// https://docs.min.io/docs/minio-kms-quickstart-guide.html
+		"MINIO_KMS_KES_ENDPOINT=" + "https://play.min.io:7373",
+		"MINIO_KMS_KES_KEY_FILE=" + "root.key",
+		"MINIO_KMS_KES_CERT_FILE=" + "root.cert",
+		"MINIO_KMS_KES_KEY_NAME=" + "my-minio-key",
+	}
+	f := e2e.NewInstrumentedRunnable(e, "loki-minio").WithPorts(ports, e2edb.AccessPortName)
+	runnable := f.Init(
+		e2e.StartOptions{
+			Image: "minio/minio:RELEASE.2022-03-03T21-21-16Z",
```

Comment:
So this hack can be removed I think once e2e is upgraded. But that can be done later in separate PR! 🙂